### PR TITLE
Add unit testing

### PR DESCRIPTION
**Pull request by alhazen11**
Screen Shot 2021-10-03 at 00 39 01

### CODE REVIEW

File: test/lib/services/projectService.test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+describe('ProjectService', () => {
+    beforeEach(() => {
+        jest.resetModules();
+        console.log = jest.fn(); // create a new mock function for each test
+    })
+    test('createNewProjectPositive', async() => {
```
Comment (retzd-tech) [inline test/lib/services/projectService.test.js]: rename to "should create new project successfully"

Comment (retzd-tech) [reply test/lib/services/projectService.test.js]: also, add new space after async

File: test/lib/services/projectService.test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        const { createNewProject } = require('@generate/lib/services/projectService');
+        const { exec } = require('child_process');
+        const projectInformation = {
+            projectType: '',
```
Comment (retzd-tech) [inline test/lib/services/projectService.test.js]: fill any relevant information, or example of the information, you can refer to the default value in the code

File: test/lib/services/projectService.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            description: '',
+            author: '',
+            repositoryUrl: ''
+        }
+
+        jest.mock('inquirer', () => ({
+            prompt: jest.fn().mockReturnValueOnce(projectInformation)
+        }))
+        jest.mock('child_process', () => ({
+            exec: jest.fn().mockImplementationOnce((command, callback) => callback(null, "ok"))
```
Comment (retzd-tech) [inline test/lib/services/projectService.test.js]: extract null into a variable first. example : 
```js
const error = null;
const log = "ok";
callback(error, log)
```

File: test/lib/services/projectService.test.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            description: '',
+            author: '',
+            repositoryUrl: ''
+        }
+
+        jest.mock('inquirer', () => ({
+            prompt: jest.fn().mockReturnValueOnce(projectInformation)
+        }))
+        jest.mock('child_process', () => ({
+            exec: jest.fn().mockImplementationOnce((command, callback) => callback("null", "ok"))
```
Comment (retzd-tech) [inline test/lib/services/projectService.test.js]: extract null into a variable first. example : 
```js
const error = {
data:  "someError"
};
const log = "ok";
callback(error, log)
```

File: test/lib/services/projectService.test.js (hunk 1 above)
Comment (alhazen11) [reply test/lib/services/projectService.test.js]: space like async () ?

**Comment by retzd-tech** [comment]
yes

File: test/lib/services/projectService.test.js (hunk 1 above)
Comment (alhazen11) [reply test/lib/services/projectService.test.js]: done

File: test/lib/services/projectService.test.js (hunk 2 above)
Comment (alhazen11) [reply test/lib/services/projectService.test.js]: done

File: test/lib/services/projectService.test.js (hunk 3 above)
Comment (alhazen11) [reply test/lib/services/projectService.test.js]: done

File: test/lib/services/projectService.test.js (hunk 4 above)
Comment (alhazen11) [reply test/lib/services/projectService.test.js]: done

**Pull request merged**